### PR TITLE
List fixes

### LINKED ISSUE #60: Query Internal Server Error
The following query causes an internal server error:
```json
{
  "message": {
    "query_graph": {
      "nodes": {
        "n0": {
          "category": "biolink:Disease",
          "id": [
            "MONDO:0005737"
          ]
        },
        "n1": {
          "category": "biolink:Gene"
        },
        "n2": {
          "category": "biolink:Disease"
        },
        "n3": {
          "category": "biolink:ChemicalSubstance"
        }
      },
      "edges": {
        "e0": {
          "subject": "n0",
          "object": "n1"
        },
        "e1": {
          "subject": "n1",
          "object": "n2"
        },
        "e2": {
          "subject": "n1",
          "object": "n3"
        },
        "e3": {
          "subject": "n3",
          "object": "n2",
          "predicate": "biolink:treats"
        }
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by alongreyber**
Allows query graphs with list properties (category, id, predicate). Fixes #60.

### CODE REVIEW

File: strider/trapi.py [hunk 1]
```diff
@@ -58,11 +58,10 @@ def get_curies(message: Message) -> list[str]:
...
     curies = set()
     if 'query_graph' in message:
-        curies |= {
-            qnode.get("id")
-            for qnode in message['query_graph']['nodes'].values()
-            if qnode.get("id", False)
-        }
+        for qnode in message['query_graph']['nodes'].values():
+            qnode_id = qnode.get("id", False)
+            if qnode_id:
```
Comment (patrickkwang) [inline strider/trapi.py:63]: This is a good example use case for the walrus operator.

File: strider/trapi.py [hunk 2]
```diff
@@ -58,11 +58,10 @@ def get_curies(message: Message) -> list[str]:
...
+                curies.update(qnode_id)
```
Comment (patrickkwang) [inline strider/trapi.py:64]: I like the new `|=` for this; it's clearer to me.

File: strider/fetcher.py [hunk 3]
```diff
@@ -150,18 +150,18 @@ async def generate_plan(self):
...
                 continue
-            result = {
-                "node_bindings": {
-                    qnode_id: [{"id": qnode["id"]}]
-                },
-                "edge_bindings": {},
-            }
-            await self.put(result)
-            break
+            for curie in qnode["id"]:
```
Comment (patrickkwang) [inline strider/fetcher.py:157]: Do we have a test for this?

File: requirements-test.txt [hunk 4]
```diff
@@ -5,6 +5,6 @@ pytest==5.4.3
 pytest-asyncio==0.14.0
 pytest-cov==2.10.0
 git+git://github.com/TranslatorIIPrototypes/kp_registry@v1.0
-git+git://github.com/ranking-agent/simple-kp@v1.0
+git+git://github.com/ranking-agent/simple-kp@master
```
Comment (patrickkwang) [inline requirements-test.txt:8]: I believe we can use the `v1.0.0` tag here.

**Review by patrickkwang — CHANGES_REQUESTED** [review]
Very nice. Minor comments to be addressed.

File: strider/trapi.py (hunk 1 above)
Comment (alongreyber) [reply strider/trapi.py:63]: How should this use the walrus operator? Do you mean something like this:
```python
qnode_id |= qnode.get("id", None)
```

File: strider/fetcher.py (hunk 3 above)
Comment (alongreyber) [reply strider/fetcher.py:157]: What do you mean exactly? This line has coverage from a couple of the tests in tests/server/test_server.py. Is there a specific condition you had in mind to test?

File: strider/trapi.py (hunk 1 above)
Comment (patrickkwang) [reply strider/trapi.py:63]: `if qnode_id := qnode.get("id", False):` I think.
https://www.python.org/dev/peps/pep-0572/

File: strider/fetcher.py (hunk 3 above)
Comment (patrickkwang) [reply strider/fetcher.py:157]: I mean a test for a qnode with multiple `id`s. Probably it's tied in with the multiple-plan issues.

Comment (alongreyber) [reply strider/fetcher.py:157]: If I make one, it will be failing. Should I still add it to this commit, or wait for your multiple plan issues?

Comment (patrickkwang) [reply strider/fetcher.py:157]: That's okay, we'll add a test later.

**Review by patrickkwang — APPROVED** [review]

**Pull request merged**